### PR TITLE
Fix issue #151: bind not restarting when user uses closeBind

### PR DESCRIPTION
**Pull request by laduchesneau**
When a pdu expires and the user  returns true on closeBind, it should now call the close function with the proper state. This will allow OnClosed to be called and the bind to be restarted.

This also fixes the the transceiver context that is never closed.

### CODE REVIEW

**Comment by laduchesneau** [comment]
closing it for now, the error message doesnt match the close reason. There seems to be a race condition between the unbind  start and the tcp disconnect. Ill reopen when I resolve this issue.

File: transceivable.go [hunk 1]
```diff
@@ -127,6 +129,8 @@ func (t *transceivable) Close() (err error) {
 		if t.settings.OnClosed != nil {
 			t.settings.OnClosed(ExplicitClosing)
 		}
+
+		t.wg.Wait()
 	}
 	return
 }
```
Comment (linxGnu) [inline transceivable.go:136]: Please update this function, reuse your nearly created `closing`
```go
func (t *transceivable) Close() error {
    t.closing(ExplicitClosing)
    return  nil
}
```

**Review by linxGnu — APPROVED** [review]
LGTM. Just a small comment.

I am sorry for replying late due to personal matter

File: transceivable.go (hunk 1 above)
Comment (laduchesneau) [reply transceivable.go:119]: Some tests were failing after recommended change. So I added a returned error to the new closing () func.
```golang
func (t *transceivable) Close() (err error) {
	return t.closing(ExplicitClosing)
}

func (t *transceivable) closing(state State) (err error) {
	if atomic.CompareAndSwapInt32(&t.aliveState, Alive, Closed) {
		...
		// close underlying conn
		err = t.conn.Close()
		...
	}
	return
}
```

**Comment by linxGnu** [comment]
@laduchesneau Can I start the final review on the PR?

**Review by linxGnu — APPROVED** [review]
Look awesome to me!!!! Thank you for the GREAT PR @laduchesneau !!!
Truly love to hear that you ran this version for months.

**Pull request merged**